addons/test_mass_mailing/tests/test_blacklist_behavior.py

```diff
@@ -16,10 +18,12 @@ def test_mailing_bounce_w_auto_bl(self):
         mailing.write({'mailing_domain': [('id', 'in', target.ids)]})
 
         # create bounced history of 4 statistics
+        traces = self.env['mailing.trace']
         for idx in range(4):
             new_mailing = mailing.copy()
-            self._create_bounce_trace(new_mailing, target, dt=datetime.datetime.now() - datetime.timedelta(weeks=idx+2))
-            self.gateway_mail_bounce(new_mailing, target)
+            traces += self._create_bounce_trace(new_mailing, target, dt=datetime.datetime.now() - datetime.timedelta(weeks=idx+2))
+            with freeze_time(datetime.datetime.now() - datetime.timedelta(weeks=idx+2)):
```

Comment:
Isn't it safer to store "datetime.datetime.now()" (or even the whole computed datetime) into a variable instead of calling it twice?